lib/jekyll-redirect-from/redirect_page.rb

```diff
@@ -1,42 +1,70 @@
 # Encoding: utf-8
-
 module JekyllRedirectFrom
+  # Specialty page which implements the redirect path logic
   class RedirectPage < Jekyll::Page
-    # Initialize a new RedirectPage.
+    # Use Jekyll's native absolute_url filter
+    include Jekyll::Filters::URLFilters
+
+    DEFAULT_DATA = {
+      "sitemap" => false,
+      "layout"  => "redirect"
+    }.freeze
+
+    # Creates a new RedirectPage instance from a source path and redirect path
     #
-    # site - The Site object.
-    # base - The String path to the source.
-    # dir  - The String path between the source and the file.
-    # name - The String filename of the file.
-    def initialize(site, base, dir, name)
-      @site = site
-      @base = base
-      @dir  = dir
-      @name = name
-
-      self.process(name)
-      self.data = { "layout" => nil, "sitemap" => false }
-
-      data.default_proc = proc do |_, key|
-        site.frontmatter_defaults.find(File.join(dir, name), type, key)
-      end
+    # site - The Site object
+    # from - the (URL) path, relative to the site root to redirect from
+    # to   - the relative path or URL which the page should redirect to
+    def self.from_paths(site, from, to)
+      page = RedirectPage.new(site, site.source, "", "redirect.html")
+      page.set_paths(from, to)
+      page
+    end
+
+    # Creates a new RedirectPage instance from the path to the given doc
+    def self.redirect_from(doc, path)
+      RedirectPage.from_paths(doc.site, path, doc.url)
+    end
+
+    # Creates a new RedirectPage instance from the doc to the given path
+    def self.redirect_to(doc, path)
+      RedirectPage.from_paths(doc.site, doc.url, path)
+    end
+
+    # Overwrite the default read_yaml method since the file doesn't exist
+    def read_yaml(_base, _name, _opts = {})
+      self.content = ""
+      self.data ||= DEFAULT_DATA.dup
```

Comment:
Why `||=`? It seems like it should either be just an assignment or a merge.

When would `self.data` already be set at this point?